api/client/contextdialer.go

```diff
@@ -52,6 +52,18 @@ func NewDirectDialer(keepAlivePeriod, dialTimeout time.Duration) ContextDialer {
 	}
 }
 
+// NewDialer makes a new dialer that connects to an Auth server either directly or via an HTTP proxy, depending
+// on the environment.
+func NewDialer(keepAlivePeriod, dialTimeout time.Duration) ContextDialer {
+	dialer := NewDirectDialer(keepAlivePeriod, dialTimeout)
+	return ContextDialerFunc(func(ctx context.Context, network, addr string) (net.Conn, error) {
+		if proxyAddr := GetProxyAddress(addr); proxyAddr != "" {
```

Comment:
nit: push to the smaller scope?

```suggestion
	return ContextDialerFunc(func(ctx context.Context, network, addr string) (net.Conn, error) {
				dialer := NewDirectDialer(keepAlivePeriod, dialTimeout)
				if proxyAddr := GetProxyAddress(addr); proxyAddr != "" {
```